app/controllers/items_controller.rb

```diff
@@ -1,19 +1,39 @@
 class ItemsController < ApplicationController
   before_action :authenticate_user!
-  before_action :set_wishlist_and_item, only: %i(edit update)
+  before_action :set_wishlist, only: %i(new create edit update)
+  before_action :set_item, only: %i(edit update)
+
+  def new
+    @item = @wishlist.items.new
+    authorize! :new, @item
+  rescue CanCan::AccessDenied
+    redirect_to @wishlist, alert: t("generic_error")
```

Comment:
in Application controller you can:
```ruby
class ApplicationController < ActionController::Base
  rescue_from ActionController::RoutingError, :with => :error_render_method

  def error_render_method
    respond_to do |type|
      type.xml { render :template => "errors/error_404", :status => 404 }
      type.all  { render :nothing => true, :status => 404 }
    end
    true
  end
end
```
and catch all there (ps for cancan there is already global method (helper) to catch all errors in one place)